### DISCUSSION TITLE
Dynamic routing with custom server

### DISCUSSION THREAD

**Discussion by shok333** [vercel/next.js#12276] [Help]
Hello. I use next.js with express.js in my application. I wanna use dynamic page "/pages/[post]" and "/pages/post1". When i click on link with "/pages/post1" application make request "/pages/post1.js" with response 404. After, my application reload page and i get "/pages/post1". How can i use dynamic page without request "/pages/post1.js" and reloading page?

**Comment by serge20** [comment]
Can you share your code snippet? You should be able to use that by using Link component and passing the proper href and as props to it.  You can look at the docs here: https://nextjs.org/docs/api-reference/next/link#dynamic-routes

```
<Link href="/post/[pid]" as="/post/abc">
  <a>First Post</a>
</Link>
```

P.S: Make sure that your file is following the proper format [post].js when you create it.
